contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -3222,3 +3222,372 @@ BEGIN
 END;
 $$
 LANGUAGE pltsql STABLE;
+
+-- sp_babelfish_autoformat is a helper procedure which formats the contents of a table (or view)
+-- as narrowly as possible given its actual column contents.
+-- This proc is currently only used by sp_who but could be applied more generically.
+-- A complication is that the metadata from #tmp tables cannot be found in the babelfish
+-- catalogs, so we have to use some trickery to make things work.
+-- Not all datatypes are handled as well as might be possible, but it is sufficient for 
+-- the current purposes.
+-- Note that this proc may increase the response time for the first execution of sp_who, but 
+-- we're looking at prioritizing user-friendliness (easy-to-read output) here. Also, sp_who 
+-- is very unlikely to be part of performance-critical workload.
+CREATE OR REPLACE PROCEDURE sys.sp_babelfish_autoformat(
+	IN "@tab"        VARCHAR(257) DEFAULT NULL,
+	IN "@orderby"    VARCHAR(1000) DEFAULT '',
+	IN "@printrc"    sys.bit DEFAULT 1,
+	IN "@hiddencols" VARCHAR(1000) DEFAULT NULL)
+LANGUAGE 'pltsql'
+AS $$
+BEGIN
+	SET NOCOUNT ON
+	DECLARE @rc INT
+	DECLARE @id INT
+	DECLARE @objtype VARCHAR(2)	
+	DECLARE @msg VARCHAR(200)	
+	
+	IF @tab IS NULL
+	BEGIN
+		RAISERROR('Must specify table name', 16, 1)
+		RETURN		
+	END
+	
+	IF TRIM(@tab) = ''
+	BEGIN
+		RAISERROR('Must specify table name', 16, 1)
+		RETURN		
+	END	
+	
+	-- Since we cannot find #tmp tables in the Babelfish catalogs, we cannot check 
+	-- their existence other than by trying to select from them
+	-- NB: not handling uncommon but valid syntax '<schemaname>.#tmp' for #tmp tables
+	IF SUBSTRING(@tab,1,1) <> '#'
```

Comment:
Function sys.babelfish_get_enr_list() could be used to determine if a #tmp table exists but the columns and datatypes can still not be retrieved, so it would be of little use here as the columns and datatypes could still not be retrieved. I added a comment to the code to mention this.
In case a #tmp table is specified which does not exist, the current code raises the same error message as we would be able to raise otherwise.